src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -150,6 +151,21 @@ public DatabaseDocumentActions(DocumentDatabase database, BuildVersionType build
 
             public void WriteDocument(DocumentItem item, SmugglerProgressBase.CountsWithLastEtag progress)
             {
+                if (item.Document.NonPersistentFlags.Contain(NonPersistentDocumentFlags.FromSmuggler) &&
+                     ((item.Document.Flags & DocumentFlags.HasRevisions) != 0) &&
+                    ((item.Document.Flags & DocumentFlags.Revision) !=0) || 
+                     ((item.Document.Flags & DocumentFlags.DeleteRevision) != 0))
+                {
+                    using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                    using (context.OpenReadTransaction())
```

Comment:
Can we avoid opening a ReadTransaction for every call?
It would be better to open transaction when we haven't opened it before and in HandleBatchOfDocumentsIfNecessary when we are actually adding the batch for processing, we can close the transaction